core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -152,4 +159,39 @@ public SparkSqlDialect(SqlDialect.Context context) {
     }
     return super.getCastSpec(type);
   }
+
+  /**
+   * Rewrite SINGLE_VALUE.
+   *
+   * <pre>In Spark, RAISE_ERROR("more than one value in agg SINGLE_VALUE")
+   * will throw exception.</pre>
+   *
+   * <blockquote><pre>
+   * CASE COUNT(*)
+   * WHEN 0 THEN NULL
+   * WHEN 1 THEN MIN(&lt;result&gt;)
+   * ELSE RAISE_ERROR("more than one value in agg SINGLE_VALUE")
+   * </pre></blockquote>
+   */
+  @Override public SqlNode rewriteSingleValueExpr(SqlNode aggCall, RelDataType relDataType) {
+    final SqlNode operand = ((SqlBasicCall) aggCall).operand(0);
```

Comment:
I don't know if you care to preserve the position of the aggCall in the constructed tree.

Reply:
The current conversion here is consistent with other dialect conversions, so no adjustments have been made.